tests/dialects/test_stencil.py

```diff
@@ -560,3 +570,74 @@ def test_stencil_access():
     assert isinstance(access, AccessOp)
     assert access.offset == offset_index_attr
     assert access.temp.typ == temp_type
+
+
+# TODO: Move to a notebook at some point with proper documentation
+def test_1d3pt_stencil_construct():
+    """
+    An example 1d-3pt stencil implementation from XDSL Python-land
+    using the stencil dialect
+    """
+
+    shape = (8,)
+    space_order = 2
+    r = space_order // 2
+    symbolic_shape = (0 - r, shape[0] - r)
```

Comment:
shouldn't this be `shape[0] + r`?

Reply:
Here the shape is the halo including size! Maybe the halo excluding size could be clearer, I'm not sure;
Though this is what George came up with and he's amongst the target stencil experts 😛

Reply:
What we need here is the correct input in the FieldType, i.e. the computational domain size. So I guess Anton is correct. Please correct me if I am wrong. @PapyChacal, you are the stencil dialect expert. What is the correct way to put this here?

Reply:
The fieldtype is none of that, it's just the buffer size. (E.g. in DeVito it's compute domain + 2 in all directions, even where the input with halo is +1)
What is expressed here is an input buffer bounded by [-1,7], which yields size 8, which is what I read by shape = (8,) personally.

You set the store bounds to [0,6], those will be the compute bounds here. The offsets are -1/+1 so the buffer size is exactly the minimum needed.

So, this is a correct stencil computation, I don't know if it's what you wanted to express though